### PR TITLE
Unify implementation of matching algorithm

### PR DESCRIPTION
**Pull request by lochel**

### CODE REVIEW

File: OMCompiler/Compiler/Util/Flags.mo [hunk 1]
```diff
@@ -853,27 +853,8 @@ constant ConfigFlag CHEAPMATCHING_ALGORITHM = CONFIG_FLAG(13, "cheapmatchingAlgo
     Util.gettext("Sets the cheap matching algorithm to use. A cheap matching algorithm gives a jump start matching by heuristics."));
 
 constant ConfigFlag MATCHING_ALGORITHM = CONFIG_FLAG(14, "matchingAlgorithm",
-  NONE(), EXTERNAL(), STRING_FLAG("PFPlusExt"),
```
Comment (kabdelhak) [inline OMCompiler/Compiler/Util/Flags.mo]: We should have a look at jens' paper where he compared all these matching algorithms. Maybe there are special cases which we are not considering right now. If not this seems superfluous.

**Comment by kabdelhak** [comment]
Did you stop jenkins or did it crash?

File: OMCompiler/Compiler/BackEnd/Matching.mo [hunk 2]
```diff
@@ -88,15 +70,29 @@ public function RegularMatching "
...
   output array<Integer> ass2 "var := ass2[eqn]";
+  output Boolean outPerfectMatching;
+algorithm
+  ass1 := arrayCreate(nVars, -1);
+  ass2 := arrayCreate(nEqns, -1);
+
+  (ass1, ass2, outPerfectMatching) := ContinueMatching(m, nVars, nEqns, ass1, ass2);
+end RegularMatching;
+
+public function ContinueMatching "
```
[the remark this answers is not in the document]
Comment (lochel) [reply OMCompiler/Compiler/BackEnd/Matching.mo]: **PerfectMatching**: This function fails if there is no perfect matching for the given system.
**RegularMatching**: This function returns at least a partial matching, starting from scratch. Unmatched nodes are represented by -1.
**ContinueMatching**: This function returns at least a partial matching, starting from a given partial matching. Unmatched nodes are represented by -1.

File: OMCompiler/Compiler/BackEnd/IndexReduction.mo [hunk 3]
```diff
@@ -2096,12 +2084,9 @@ algorithm
...
         // match the variables not the equations, to have preferred states unmatched
-        vec1 = Array.expand(nfreeStates,ass1,-1);
-        vec2 = Array.expand(neqns,ass2,-1);
-        true = BackendDAEEXT.setAssignment(nv1,ne1,vec1,vec2);
-        Matching.matchingExternalsetIncidenceMatrix(ne1, nv1, mT1);
-        BackendDAEEXT.matching(ne1, nv1, 3, -1, 0.0, 0);
-        BackendDAEEXT.getAssignment(vec1, vec2);
+        vec1 = Array.expand(nfreeStates, ass1, -1);
+        vec2 = Array.expand(neqns, ass2, -1);
+        (vec1, vec2, _) = Matching.ContinueMatching(mT1, nv1, ne1, vec1, vec2);
```
Comment (lochel) [inline OMCompiler/Compiler/BackEnd/IndexReduction.mo]: @kabdelhak Should I swap vec1 and vec2 here?

File: OMCompiler/Compiler/BackEnd/IndexReduction.mo [hunk 4]
```diff
@@ -2135,11 +2120,7 @@ algorithm
         ne = BackendEquation.equationArraySize(eqns);
         mT = AdjacencyMatrix.transposeAdjacencyMatrix(m,nv);
         // match the variables not the equations, to have preferred states unmatched
-        Matching.matchingExternalsetIncidenceMatrix(ne,nv,mT);
-        BackendDAEEXT.matching(ne,nv,3,-1,1.0,1);
-        vec1 = arrayCreate(nv,-1);
-        vec2 = arrayCreate(ne,-1);
-        BackendDAEEXT.getAssignment(vec1,vec2);
+        (vec1, vec2, _) = Matching.RegularMatching(mT, nv, ne);
```
Comment (lochel) [inline OMCompiler/Compiler/BackEnd/IndexReduction.mo]: ... and here?

File: OMCompiler/Compiler/BackEnd/Tearing.mo [hunk 5]
```diff
@@ -735,11 +734,7 @@ algorithm
   map := listArray(maplst);
   // get for each residual a tvar
   size := arrayLength(map);
-  Matching.matchingExternalsetIncidenceMatrix(size,size,map);
-  BackendDAEEXT.matching(size,size,5,-1,1.0,1);
-  v1 := arrayCreate(size,-1);
-  v2 := arrayCreate(size,-1);
-  BackendDAEEXT.getAssignment(v2,v1);
+  (v2, v1, _) := Matching.RegularMatching(map, size, size);
```
Comment (lochel) [inline OMCompiler/Compiler/BackEnd/Tearing.mo]: ... and here?

**Comment by lochel** [comment]
This didn't work out so well, unfortunately.

**Pull request closed without merging**